config/raw_config.go

```diff
@@ -93,6 +93,8 @@ func (r *RawConfig) Value() interface{} {
 // structure will always successfully decode into its ultimate
 // structure using something like mapstructure.
 func (r *RawConfig) Config() map[string]interface{} {
+	r.lock.Lock()
+	defer r.lock.Unlock()
```

Comment:
Relevant questions:

 * Do callers require write access to the result returned?
 * If yes, should we do a copy here?